src/SchemaReader.php

```diff
@@ -2,540 +2,633 @@
+    protected function loadSequenceChildNode(
+        ElementContainer $elementContainer,
+        DOMElement $node,
+        DOMElement $childNode,
+        $max
+    ) {
+        $commonMethods = [
+            [
+                ['sequence', 'choice', 'all'],
+                [$this, 'loadSequenceChildNodeLoadSequence'],
+                [
+                    $elementContainer,
+                    $childNode,
+                    $max,
+                ],
+            ],
+        ];
+        $methods = [
+            'element' => [
+                [$this, 'loadSequenceChildNodeLoadElement'],
+                [
+                    $elementContainer,
+                    $node,
+                    $childNode,
+                    $max,
+                ],
+            ],
+            'group' => [
+                [$this, 'loadSequenceChildNodeLoadGroup'],
+                [
+                    $elementContainer,
+                    $node,
+                    $childNode,
+                ],
+            ],
+        ];
+
+        $this->maybeCallCallableWithArgs($childNode, $commonMethods, $methods);
     }
 
-    private function loadElementRef(ElementDef $referenced, DOMElement $node)
-    {
-        $ref = new ElementRef($referenced);
-        $ref->setDoc($this->getDocumentation($node));
+    /**
+     * @param int|null $max
+     */
+    protected function loadSequenceChildNodeLoadSequence(
+        ElementContainer $elementContainer,
+        DOMElement $childNode,
+        $max
+    ) {
+        $this->loadSequence($elementContainer, $childNode, $max);
+    }
 
-        if ($node->hasAttribute('maxOccurs')) {
-            $ref->setMax($node->getAttribute('maxOccurs') == 'unbounded' ? -1 : (int) $node->getAttribute('maxOccurs'));
-        }
-        if ($node->hasAttribute('minOccurs')) {
-            $ref->setMin((int) $node->getAttribute('minOccurs'));
-        }
-        if ($node->hasAttribute('nillable')) {
-            $ref->setNil($node->getAttribute('nillable') == 'true');
+    /**
+     * @param int|null $max
+     */
+    protected function loadSequenceChildNodeLoadElement(
+        ElementContainer $elementContainer,
+        DOMElement $node,
+        DOMElement $childNode,
+        $max
+    ) {
+        if ($childNode->hasAttribute('ref')) {
+            /**
+             * @var ElementDef $referencedElement
+             */
+            $referencedElement = $this->findSomething('findElement', $elementContainer->getSchema(), $node, $childNode->getAttribute('ref'));
+            $element = static::loadElementRef(
+                $referencedElement,
+                $childNode
+            );
+        } else {
+            $element = $this->loadElement(
+                $elementContainer->getSchema(),
+                $childNode
+            );
         }
-        if ($node->hasAttribute('form')) {
-            $ref->setQualified($node->getAttribute('form') == 'qualified');
+        if (is_int($max) && (bool) $max) {
```

Comment:
should this be `is_int($max) && $max>1` ?

Reply:
I believe this comes from [line 296](https://github.com/goetas-webservices/xsd-reader/blob/bb953a590c79e0f8e800e913a50768a40ed9daa1/src/SchemaReader.php), was `if ($max)`.

I'll check it tonight with both  `$max >= 1` & `$max > 1`- although do note that `(bool) -1` is true, I don't think it comes up practically?

Reply:
tests pass with both >= 1 and > 1, putting it in as > 1 as suggested.